Utils/AsyncTaskList.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace DUCK.Utils
+{
+	public class AsyncTaskList
+	{
+		private readonly List<Action<Action>> tasks = new List<Action<Action>>();
+		private int currentTaskIndex;
+
+		/// <summary>
+		/// Adds an async task to the task list. The Action provided takes another action called next.
+		/// Call this next function to indicate the task is complete and the next one in the chain, should be called
+		/// Eg:
+		/// taskList.Add(next => {
+		///   // do some work, then call next when complete.
+		///   next();
+		/// });
+		/// </summary>
+		/// <param name="doWork">The task function</param>
+		public void Add(Action<Action> doWork)
```

Comment:
Not a huge fan of the name `doWork` but I can't think of a good one

Reply:
executeTask i guess is better

Reply:
or just `execute`